pkg/controller/networkpolicy/antreanetworkpolicy.go

```diff
@@ -123,38 +124,56 @@ func (n *NetworkPolicyController) deleteANP(old interface{}) {
 // in case of ADD event or modified and store the updated instance, in case
 // of an UPDATE event.
 func (n *NetworkPolicyController) processAntreaNetworkPolicy(np *secv1alpha1.NetworkPolicy) *antreatypes.NetworkPolicy {
-	appliedToGroupNames := make([]string, 0, len(np.Spec.AppliedTo))
-	// Create AppliedToGroup for each AppliedTo present in
-	// AntreaNetworkPolicy spec.
+	appliedToPerRule := np.Spec.AppliedTo == nil
+	// appliedToGroupNames tracks all distinct appliedToGroups referred to by the Antrea NetworkPolicy,
+	// either in the spec section or in ingress/egress rules.
```

Comment:
I think it would be worth adding the information you gave in the PR description to this comment as well, as the reason for setting AppliedToGroups at the internal policy level even when rules have their own AppliedToGroups:

> For the internalNP, AppliedToGroups are set as spec.appliedTo or the union of all rule.appliedTo. The original span calculations and stale appliedToGroup cleanups would seamlessly work.

Reply:
Sure.